We will be using SVC (Support Vector Classifier) from the Scikit-learn library to implement this model and evaluating the predictions using a classification report and a confusion matrix. We will later optimize the model using grid search.

class  cap-diameter cap-shape cap-surface cap-color  \
0         p         15.26         x           g         o   
1         p         16.60         x           g         o   
2         p         14.07         x           g         o   
3         p         14.17         f           h         e   
4         p         14.64         x           h         o   
...     ...           ...       ...         ...       ...   
61064     p          1.18         s           s         y   
61065     p          1.27         f           s         y   
61066     p          1.27         s           s         y   
61067     p          1.24         f           s         y   
61068     p          1.17         s           s         y   

      does-bruise-or-bleed gill-attachment gill-color  stem-height  \
0                        f               e          w        16.95   
1                        f               e          w        17.99   
2                        f               e          w        17.80   
3                        f               e          w        15.77   
4                        f               e          w        16.53   
...                    ...             ...        ...          ...   
61064                    f               f          f         3.93   
61065                    f               f          f         3.18   
61066                    f               f          f         3.86   
61067                    f               f          f         3.56   
61068                    f               f          f         3.25   

       stem-width stem-color has-ring ring-type habitat season  
0           17.09          w        t         g       d      w  
1           18.19          w        t         g       d      u  
2           17.74          w        t         g       d      w  
3           15.98          w        t         p       d      w  
4           17.20          w        t         p       d      w  
...           ...        ...      ...       ...     ...    ...  
61064        6.22          y        f         f       d      a  
61065        5.43          y        f         f       d      a  
61066        6.37          y        f         f       d      u  
61067        5.44          y        f         f       d      u  
61068        5.45          y        f         f       d      u  

[61069 rows x 15 columns]

In [40]:
##import libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.model_selection import GridSearchCV

##convert non-numeric variables
df1 = pd.read_csv('Mushroom_clean.csv')



from sklearn.preprocessing import LabelEncoder

# Define a list of features you want to encode
features_to_encode = ['class', 'cap-shape','cap-surface','cap-color','does-bruise-or-bleed','gill-attachment','gill-color','stem-color','has-ring','ring-type','habitat','season']

# Initialize the LabelEncoder
label_encoder = LabelEncoder()

# Encode each feature in the DataFrame
for feature in features_to_encode:
    df1[feature] = label_encoder.fit_transform(df1[feature])
df1
# If you want to transform new data using the same encoder,
##Set x and y variables
df2=df1.head(10000)
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   class                 10000 non-null  int64  
 1   cap-diameter          10000 non-null  float64
 2   cap-shape             10000 non-null  int64  
 3   cap-surface           10000 non-null  int64  
 4   cap-color             10000 non-null  int64  
 5   does-bruise-or-bleed  10000 non-null  int64  
 6   gill-attachment       10000 non-null  int64  
 7   gill-color            10000 non-null  int64  
 8   stem-height           10000 non-null  float64
 9   stem-width            10000 non-null  float64
 10  stem-color            10000 non-null  int64  
 11  has-ring              10000 non-null  int64  
 12  ring-type             10000 non-null  int64  
 13  habitat               10000 non-null  int64  
 14  season                10000 non-null  int64  
dtypes: float64(3), int64

In [44]:
x=df2.drop('class', axis=1)
y=df2['class']

##Split the data into training and test tests
x_train, x_test, y_train, y_test =  train_test_split(x,y,test_size=0.3, random_state=12)

##set algorithm
model=SVC()

##fit the algorith to the training data
model.fit(x_train, y_train)

#assign a new variable to the prediction of the x test data
model_predict=model.predict(x_test)

#confusion matrix
print(confusion_matrix(y_test, model_predict))

#classification report
print(classification_report(y_test, model_predict))




[[1493    3]
 [  30 1474]]
              precision    recall  f1-score   support

           0       0.98      1.00      0.99      1496
           1       1.00      0.98      0.99      1504

    accuracy                           0.99      3000
   macro avg       0.99      0.99      0.99      3000
weighted avg       0.99      0.99      0.99      3000



The confusion matrix reports a low occurance of false-positives (3), and the classification report states that precision, recall, and the f1-score are all above 0.97

Dear mr. Anderson The above Support vector matrix, is being used to analyze the edibility of mushrooms based on visible traits that can be observed on the mushroom. given the degree of detail and number of features represented in the dataset we were able to yield a very accurate and scaleable confusion matrix, which is accurate up to the 97th percentile, making the data extremely valuable for predicting the edibility and classification of an unknown mushroom. It can be observed above that we've used a subset of the origional data for ease of processing, splitting a 10000 data subset, into a test and training pool. That said 7000 data points were used to train the model, and 3000 datapoints were used to test the model. in testing the model returned with precision and f1 scores both at 98% or highe